src/BloomExe/Book/BookStorage.cs

```diff
@@ -866,6 +866,8 @@ private string SanitizeNameForFileSystem(string name)
 				name = name.Replace(c, ' ');
 			}
 			name = name.Trim();
+			if (name.StartsWith("."))
```

Comment:
OK. So my aim here is to give up close-as-possible fidelity in exchange for some simplicity since we keep finding new special cases.  I played around with this... what do you think of this approach?

```
private string SanitizeNameForFileSystem(string name)
		{
			name = RemoveDangerousCharacters(name);
			if (name.Length == 0)
			{
				//the localized default book name could itself have dangerous characters
				name = RemoveDangerousCharacters(BookStarter.UntitledBookName);
			}
			
			const int MAX = 50;//arbitrary
			if(name.Length >MAX)
				return name.Substring(0, MAX);
			return name;
		}

		private static string RemoveDangerousCharacters(string name)
		{
			var dangerousCharacters = new List<char>();
			dangerousCharacters.AddRange(PathUtilities.GetInvalidOSIndependentFileNameChars());
			;
			dangerousCharacters.Add('.');
			foreach (char c in dangerousCharacters)
			{
				name = name.Replace(c, ' ');
			}
			return name.Trim();
		}
```

Using a newly factored-out access to the generic book name that is in BookStarter.

```
public static string UntitledBookName
		{
			get
			{
				return LocalizationManager.GetString("EditTab.NewBookName", "Book",
					"Default file and folder name when you make a new book, but haven't give it a title yet.");
			}
		}
```